### ISSUE TITLE
Runtime error with PGI community edition 17.10

### ISSUE THREAD

**Issue by paulromano** [openmc-dev/openmc#859]
PGI has a compiler bug that prevents OpenMC from compiling:
```
PGF90-S-0038-Symbol, size, has not been explicitly declared (/home/romano/openmc/src/stl_vector.F90)
  1 inform,   0 warnings,   1 severes, 0 fatal for clear_int
```
I've run into problems with PGI before getting confused when you have local variables defined somewhere that have the same name as an intrinsic (preventing it from being used elsewhere even though it doesn't make any sense). I'm going to file a report with PGI and look into workaround ahead of our 0.9 release.

**Comment by paulromano** [comment]
Evidently the community edition also [doesn't work](https://www.pgroup.com/userforum/viewtopic.php?t=5442&view=previous&sid=ce9240125d2b931f57417ef29279246f) for C++ if the system you're compiling on uses GNU 6.0+ C++ headers. Supposed to be fixed in version 17 which has only been released as the professional edition (not the community edition).

**Comment by paulromano** [comment]
I have a workaround for the PGF90-S-0038 error but now ran into:
```
PGF90-S-0155-Dummy argument names must be the same as those in parent's interface for type bound procedure get_next_bin_energyfunction (/home/romano/openmc/src/tally_filter.F90: 1649)
```
Doesn't make sense to me as the dummy argument names do match those of the parent's interface. I will file yet another bug report to PGI.

**Comment by paulromano** [comment]
Evidently the first compiler bug is [already fixed](http://www.pgroup.com/userforum/viewtopic.php?p=21686) in the latest release and a new community edition will be out soon.

**Comment by paulromano** [comment]
I've confirmed that both of the previous bugs mentioned are now fixed in PGI community edition 17.4. However, now that I've gotten past those issues, I've uncovered yet another bug, which I [just reported to PGI](http://www.pgroup.com/userforum/viewtopic.php?p=21785).

**Comment by paulromano** [comment]
To elaborate on the problem a little bit: the compiler chokes when reading an internal file that has newline characters in it. This causes `get_node_array_*` from `xml_interface.F90` to fail because, e.g., it can't read an array like:
```XML
<universes>
   11 10
   12 13
</universes
```

**Comment by paulromano** [comment]
The bug I reported previously in 17.4 has been [fixed in 17.10](http://www.pgroup.com/userforum/viewtopic.php?p=21785). That means that we can now at least build OpenMC using PGI, but using it is a different story. I'm getting all sorts of weird, un-debuggable segfaults and errors, and I can't get a proper traceback. I'm abandoning my efforts to compile with PGI; it's just not stable enough...
